File: idea1/Section 1/Apple_Probability.py
import math
import sys

alive_outcomes = 0
Apples_Total = int(input("Apples in Total: "))
Poisonous = int(input("Poisonous Apples: "))
Apples_To_Kill = int(input("Amount of Poisonous Apples to kill: "))
Apples_Eat = int(input("Amount of Apples You Will Eat: "))

if Apples_To_Kill > Poisonous or Apples_Eat > Apples_Total:
    print("You'll be fine ")
    sys.exit()

def binomial_coefficients(N, K):
    if N - K < 0:
        return 0
    result = (math.factorial(N)) // (math.factorial(K) * (math.factorial(N - K)))
    return result
    
for case_num in range(0, Apples_To_Kill):
    var1 = binomial_coefficients(Poisonous, case_num)
    var2 = binomial_coefficients(Apples_Total - Poisonous, Apples_Eat - case_num)
    var3 = binomial_coefficients(Apples_Total, Apples_Eat)
    Numerator = (var1) * (var2)
    Denominator = var3
    alive_outcomes += (Numerator / Denominator)

print(alive_outcomes)

File: idea1/Section 1/Apple_Probability.js
const readline = require('readline-sync');

let alive_outcomes = 0;
let total_outcomes = 0;

const Apples_Total = parseInt(readline.question("Apples in Total: "));
const Poisonous = parseInt(readline.question("Poisonous Apples: "));
const Apples_To_Kill = parseInt(readline.question("Amount of Poisonous Apples to kill: "));
const Apples_Eat = parseInt(readline.question("Amount of Apples You Will Eat: "));

if (Apples_To_Kill > Poisonous || Apples_Eat > Apples_Total) {
    console.log("You'll be fine, the apple amounts make no sense");
    process.exit();
}

function binomial_coefficients(N, K) {
    if (K > N) {
        return 0;
    }
    return factorial(N) / (factorial(K) * factorial(N - K));
}

function factorial(n) {
    if (n === 0) {
        return 1;
    }
    return n * factorial(n - 1);
}

for (let case_num = 0; case_num < Apples_To_Kill; case_num++) {
    let var1 = binomial_coefficients(Poisonous, case_num);
    let var2 = binomial_coefficients(Apples_Total - Poisonous, Apples_Eat - case_num);
    alive_outcomes += var1 * var2;
}

total_outcomes = binomial_coefficients(Apples_Total, Apples_Eat);

console.log(((alive_outcomes / total_outcomes) * 100).toFixed(2) + "%");
